openmdao/core/system.py

```diff
@@ -465,6 +467,98 @@ def fd_jacobian(self, params, unknowns, resids, total_derivs=False,
+
+                for var, val in iteritems(dunknowns.flat):
+                    # Skip all states
+                    if (gsouts is None or var in gsouts) and \
+                           var not in states:
+                        dresids.flat[var] += val
+
+            else: 
+                for val in itervalues(dparams.flat):
+                    val[:] = 0.0
+                for val in itervalues(dunknowns.flat):
+                    val[:] = 0.0
+
+                # Sign on the local Jacobian needs to be -1 before
+                # we add in the fake residual. Since we can't modify
+                # the 'du' vector at this point without stomping on the
+                # previous component's contributions, we can multiply
+                # our local 'arg' by -1, and then revert it afterwards.
+                dresids.vec *= -1.0
+                try: 
+                    if force_fd:
+                        self._apply_linear_jac(self.params, self.unknowns, dparams, dunknowns, dresids, mode)
+                    else: 
+                        self.apply_linear(self.params, self.unknowns, dparams, dunknowns, dresids, mode)
+                finally: 
+                    dparams._apply_unit_derivatives(iterkeys(dparams))
+
+                dresids.vec *= -1.0 
+                for var, val in iteritems(dresids.flat):
+                    # Skip all states
+                    if (gsouts is None or var in gsouts) and \
+                            var not in states:
+                        dunknowns.flat[var] += val
+
+    def sys_jacobian(self, params, unknowns, resids, total_derivs=False): 
+        """
+        Entry point for all callers to cause linearization 
+        of system and all children of system
+
+        Args
+        ----
+        params : `VecWrapper`
+            `VecWrapper` containing parameters. (p)
+
+        unknowns : `VecWrapper`
+            `VecWrapper` containing outputs and states. (u)
+
+        resids : `VecWrapper`
+            `VecWrapper` containing residuals. (r)
+
+        total_derivs: bool
+            flag indicating if total or partial derivatives are being requested
+            
+        """
+        if self.fd_options['force_fd']: 
+            if total_derivs: 
+                self._jacobian_cache = self.fd_jacobian(params, unknowns, resids, total_derivs=True)
+            else:
+                self._jacobian_cache = self.fd_jacobian(params, unknowns, resids, total_derivs=False)
+        else: 
+            self._jacobian_cache = self.jacobian(params, unknowns, resids)
+        
+        if self._jacobian_cache is not None:
+            jc = self._jacobian_cache
+            for key, J in iteritems(jc):
+                if isinstance(J, real_types):
+                    jc[key] = np.array([[J]])
+                shape = jc[key].shape
+                if len(shape) < 2:
+                    jc[key] = jc[key].reshape((shape[0], 1))
+
+        return self._jacobian_cache
+
```

Comment:
I see how it works with calling Super in the function in Group, so as long as you are satisfied with it, and it passes, and CADRE and Mission Allocation still work.